src/main/java/seedu/guest/model/guest/IsRoomClean.java

```diff
@@ -10,26 +10,51 @@
 public class IsRoomClean {
 
     public static final String MESSAGE_CONSTRAINTS =
-            "Is Room Clean should only contain the strings \"yes\" or \"no\"";
-    public static final String VALIDATION_REGEX = "^yes$|^no$";
+            "IsRoomClean field should only contain any case-insensitive versions of the strings"
+                    + "\"yes\", \"no\", \"y\" or \"n\".";
+    public static final String VALIDATION_REGEX = "^yes$|^no$|^y$|^n$";
+    public static final String DEFAULT_IS_ROOM_CLEAN = "yes";
     public final String value;
 
     /**
      * Constructs a {@code IsRoomClean}.
      *
-     * @param isClean A valid is room clean.
+     * @param isRoomClean A valid isRoomClean.
      */
-    public IsRoomClean(String isClean) {
-        requireNonNull(isClean);
-        checkArgument(isValidIsRoomClean(isClean), MESSAGE_CONSTRAINTS);
-        value = isClean;
+    public IsRoomClean(String isRoomClean) {
+        requireNonNull(isRoomClean);
+        String isRoomCleanToLower = isRoomClean.toLowerCase();
+        checkArgument(isValidIsRoomClean(isRoomCleanToLower), MESSAGE_CONSTRAINTS);
+        value = standardiseIsRoomClean(isRoomCleanToLower);
     }
 
     /**
-     * Returns true if a given string is a valid is room clean.
+     * Returns true if a given string is a valid isRoomClean.
      */
     public static boolean isValidIsRoomClean(String test) {
-        return test.matches(VALIDATION_REGEX);
+        String testToLower = test.toLowerCase();
+        return testToLower.matches(VALIDATION_REGEX);
+    }
+
+    /**
+     * Returns standardised notation of the isRoomClean field.
+     *
+     * @param status a valid isRoomClean.
+     */
+    private String standardiseIsRoomClean(String status) {
+        requireNonNull(status);
+        assert isValidIsRoomClean(status);
+
+        String standardisedRoomUnclean = "no";
+        String standardisedRoomClean = "yes";
+        String abbreviatedRoomUnclean = "n";
+        String unabbreviatedRoomUnclean = "no";
+
+        if (status.equals(abbreviatedRoomUnclean) || status.equals(unabbreviatedRoomUnclean)) {
+            return standardisedRoomUnclean;
```

Comment:
Aren't `standardisedRoomUnclean` and `unabbreviatedRoomUnclean` the same? Can we eliminate one of these variables?

Reply:
Yes, you are right. Initially, I had both as they served different purposes, but I think reusing the variables in this instance works.